File: fun/Actor.GameHub.Identity.EntityFrameworkCore.FSharp/IdentityDbRepository.fs
module Actor.GameHub.Identity.EntityFrameworkCore.IdentityDbRepository

open Microsoft.EntityFrameworkCore

open Actor.GameHub.Identity.Abstractions
open Actor.GameHub.Identity.EntityFrameworkCore.IdentityDbContext

type IdentityDbRepository(dbOptions: DbContextOptions<IdentityDbContext>) =
    interface IIdentityRepository with
        member this.FindUserByUsernameForAuth username =
            let dbContext = new IdentityDbContext(dbOptions)

            let userQuery =
                query {
                    for user in dbContext.User do
                        where (user.Username = username)

                        select
                            { UserId = user.Id
                              Username = user.Username }
                }

            userQuery |> Seq.tryHead

let newIdentityDbRepository dbOptions = new IdentityDbRepository(dbOptions)

File: csh/Actor.GameHub.Identity.EntityFrameworkCore/IdentityDbRepository.cs
using System.Linq;
using System.Threading;
using System.Threading.Tasks;
using Actor.GameHub.Identity.Abstractions;
using Microsoft.EntityFrameworkCore;

namespace Actor.GameHub.Identity.EntityFrameworkCore
{
  public class IdentityDbRepository : IIdentityRepository
  {
    private readonly IDbContextFactory<IdentityDbContext> _dbContextFactory;

    public IdentityDbRepository(IDbContextFactory<IdentityDbContext> dbContextFactory)
    {
      _dbContextFactory = dbContextFactory;
    }

    public async Task<UserForAuth?> FindUserByUsernameForAuthAsync(string username, CancellationToken cancellationToken = default)
    {
      var dbContext = _dbContextFactory.CreateDbContext();

      var user = await dbContext.User
        .Where(u => u.Username == username)
        .Select(u => new UserForAuth
        {
          UserId = u.Id,
          Username = u.Username,
        })
        .SingleOrDefaultAsync(cancellationToken)
        .ConfigureAwait(false);

      return user;
    }
  }
}
